src/api/app/models/kiwi/image.rb

```diff
@@ -104,16 +104,17 @@ def to_xml
 
     return nil unless image && image.first_element_child
 
-    doc.xpath("image/packages").remove
-    xml_packages = package_groups.map(&:to_xml).join("\n")
-    image.first_element_child.after(xml_packages)
+    # for now we only write the default package group
+    xml_packages = default_package_group.to_xml
+    packages = doc.xpath('image/packages[@type="image"]').first
+    packages ? packages.replace(xml_packages) : image.last_element_child.after(xml_packages)
 
+    previous = doc.xpath("image/repository").first.try(:previous) || image.last_element_child
```

Comment:
I prefer to use another name for this variable... something like ```repository_position``` because in the case of the ```image.last_element_child``` this is not the previous node of anything.

Reply:
Is this explicitly tested?